app/views/vaults/show.html.erb

```diff
@@ -26,13 +26,13 @@
       <% if @interaction_enabled %>
       <div class="overlay">
         <p>
-          <%= link_to "Show", vault_snippet_path(@vault.id, snippet.id), class: "btn btn-default btn-xs" %>
-          <%= link_to "Edit", edit_vault_snippet_path(@vault.id, snippet.id), class: "btn btn-default btn-xs" %>
-          <%= link_to "Delete", vault_snippet_path(@vault.id, snippet.id), method: :delete, data: { confirm: 'Are you sure?' }, class: "btn btn-danger btn-xs" %>
+          <%= link_to "Show", vault_snippet_path(@vault.id, snippet.id, username: @vault.user.username), class: "btn btn-default btn-xs" %>
+          <%= link_to "Edit", edit_vault_snippet_path(@vault.id, snippet.id, username: @vault.user.username), class: "btn btn-default btn-xs" %>
```

Comment:
@mrvncaragay  Using `@vault.username` would be better here, given that you already delegated `#username` through to `User` in `models/vault.rb`